src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -263,7 +263,9 @@ internal class CSharpRecommendationService : AbstractRecommendationService
                 : context.SemanticModel.LookupSymbols(context.LeftToken.SpanStart);
 
             // Filter out any extension methods that might be imported by a using static directive.
-            symbols = symbols.Where(symbol => !symbol.IsExtensionMethod());
+            // But include extension methods declared in the context's type
+            symbols = symbols.Where(symbol => !symbol.IsExtensionMethod()
+            || symbol.ContainingType.DeclaringSyntaxReferences.Any(x => x.GetSyntax().IsEquivalentTo(context.ContainingTypeDeclaration))).ToList();
```

Comment:
What's the purpose of that added call to `ToList`? This code was working without that before. Why include that allocation?